### PR TITLE
feat: Delete team members except earliest admin on subscription cancellation

### PR DESCRIPTION
**Pull request by gentamura**
## Summary
Implement logic to automatically remove team members (except the earliest admin) when a subscription is canceled, ensuring teams revert to the free plan's single-member limitation.

## Related Issue
N/A

## Changes
- Add CancellationNotice component to display subscription end date
- Implement consistent return interface for getSubscription
- Add subscription status check in deleteSubscription function
- Implement team member cleanup logic:
  - Identify the earliest admin member using team_memberships table
  - Delete all other team members while preserving the earliest admin
- Use type-safe database queries with explicit limit(1)
- Add subscription cancellation invoice processing skip logic
- Add CancellationNotice component to show subscription end date
- Add error handling for subscription fetching

## Testing
1. Create a team with multiple members (admins and regular members)
2. Cancel the team's subscription
3. Verify only the earliest admin remains in the team
4. Verify other members are successfully removed

## Other Information
- Full invoice processing for subscription cancellation will be implemented separately

### CODE REVIEW

**Comment by gentamura** [comment]
@satococoa review, please :pray:

File: app/(main)/settings/team/billing-section.tsx [hunk 1]
```diff
@@ -37,6 +46,30 @@ export default async function BillingSection() {
 	);
 }
 
+type CancellationNoticeProps = {
+	subscriptionId: string;
+};
+
+async function CancellationNotice({ subscriptionId }: CancellationNoticeProps) {
```
Comment (satococoa) [inline app/(main)/settings/team/billing-section.tsx:53]: 👍

File: app/webhooks/stripe/route.ts [hunk 2]
```diff
@@ -72,8 +72,39 @@ export async function POST(req: Request) {
...
+						"The checkout session is missing a valid customer ID. Please check the session data.",
+					);
+				}
+				await deleteSubscription(event.data.object.id);
+				break;
+
 			case "invoice.created":
 				console.log(`🔔  Invoice created: ${event.data.object.id}`);
+
+				// TODO: Skip for now - will be handled when implementing subscription cancellation invoice processing
```
Comment (satococoa) [inline app/webhooks/stripe/route.ts:90]: 👍

File: services/external/stripe/actions/delete-subscription.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import { db } from "@/drizzle/db";
+import { subscriptions, teamMemberships } from "@/drizzle/schema";
+import { and, eq, ne } from "drizzle-orm";
+import { stripe } from "../config";
+
+export async function deleteSubscription(subscriptionId: string) {
```
Comment (satococoa) [inline services/external/stripe/actions/delete-subscription.ts]: This method's name is a little bit misleading.
I think `handleSubscriptionDeletion` or similar is better here, because this method won't remove subscription.

File: services/external/stripe/actions/delete-subscription.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	const subscription = await stripe.subscriptions.retrieve(subscriptionId);
+
+	if (subscription.status === "canceled") {
```
Comment (satococoa) [inline services/external/stripe/actions/delete-subscription.ts]: I suggest using an early return pattern here to reduce nesting and improve readability:

```
if (subscription.status !== "canceled") {
  return;
}
```

File: services/external/stripe/actions/delete-subscription.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		// Get the team_db_id from subscriptions table
+		const [sub] = await db
+			.select({ teamDbId: subscriptions.teamDbId })
+			.from(subscriptions)
+			.where(eq(subscriptions.id, subscriptionId))
+			.limit(1);
+
+		if (sub) {
```
Comment (satococoa) [inline services/external/stripe/actions/delete-subscription.ts]: Could we handle these cases as exceptions? While these scenarios shouldn't occur in normal operation, I think we should throw exceptions to be notified if they do happen, rather than silently continuing:
- subscription not found
- earliestAdmin not found

This would help us detect any potential issues early. What do you think?

**Review by satococoa — CHANGES_REQUESTED** [review]
Good! Thank you 👍
I've added some review comments. Could you take a look?

File: app/(main)/settings/team/billing-section.tsx (hunk 1 above)
Comment (gentamura) [reply app/(main)/settings/team/billing-section.tsx:53]: updated 4a7200794a53 ♻️

File: services/external/stripe/actions/delete-subscription.ts (hunk 3 above)
Comment (gentamura) [reply services/external/stripe/actions/delete-subscription.ts]: updated 4a7200794a53 8cf1a4f71043 ♻️

File: services/external/stripe/actions/delete-subscription.ts (hunk 4 above)
Comment (gentamura) [reply services/external/stripe/actions/delete-subscription.ts]: updated 42c0a6ddbc0f ♻️

File: services/external/stripe/actions/delete-subscription.ts (hunk 5 above)
Comment (gentamura) [reply services/external/stripe/actions/delete-subscription.ts]: updated c7a5f3ae0cb0 ♻️

**Review by satococoa — APPROVED** [review]
👍 👍 Thank you!

**Comment by gentamura** [comment]
@shige @toyamarinyon review, please :pray:

File: services/external/stripe/actions/upsert-subscription.ts [hunk 6]
```diff
@@ -12,6 +12,7 @@ const timestampToDateTime = (timestamp: number) => new Date(timestamp * 1000);
 
 export const upsertSubscription = async (subscriptionId: string) => {
 	const subscription = await stripe.subscriptions.retrieve(subscriptionId);
+
```
Comment (toyamarinyon) [inline services/external/stripe/actions/upsert-subscription.ts]: Unnecessary changes?

**Review by toyamarinyon — APPROVED** [review]

**Review by shige — APPROVED** [review]
LGTM 👍 

---

📝 For My Note

If the requirements cannot be met with this specification, we will consider adding an owner role.

File: services/external/stripe/actions/upsert-subscription.ts (hunk 6 above)
Comment (gentamura) [reply services/external/stripe/actions/upsert-subscription.ts]: Removed [3c53547](https://github.com/giselles-ai/giselle/pull/263/commits/3c53547ce7cf19c303676e21015a9c7982195821)  🗑️

**Comment by gentamura** [comment]
Thanks for reviewers 🚀
